core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java

```diff
@@ -108,11 +104,25 @@ private GenericTypes() {
      * @param type the type to be resolved in the scope of <code>contextType</code>
      * @param contextType the context type in which <code>type</code> is interpreted to resolve the type.
      * @return the resolved type.
+     * @deprecated use {@link #resolveType(TypeVariable, Type)} instead
      */
+    @SuppressWarnings("unchecked")
+    @Deprecated
     public static Type resolveType(Type type, Type contextType) {
-        return TypeToken.of(contextType)
-                .resolveType(type)
-                .getType();
+        return resolveType((TypeVariable<? extends Class<?>>) type, contextType);
```

Comment:
We need to have a backup plan here if `type` isn't a `TypeVariable`.

Reply:
100% of the uses so far are of `TypeVariable`s, there are no uses of this method.  I only left it so as not to break the public API.  I think it might have been a mistake to have it take `Type` in the first place not `TypeVariable`.

Reply:
Oh, or maybe this is my place to have it emulate the behavior of Guava.

Reply:
> 100% of the uses so far are of `TypeVariable`s

I assumed that the only reason for the changes in calculating return types in SQL Object was because this stopped working under GeAnTyRef (god that's awful to type out)

Reply:
Yeah, you're right, I am wrong.  I made that change because the name is nicer but didn't realize it was hiding this behavior change.